### PR TITLE
133.fix alert dialog edit delete

### LINKED ISSUE #133: Fix AlertDialog on Long Hold for Edit/Delete
[//]: # (These are comments and are used for reference and will not show up in the PR)

User Requirement: N/A

[//]: # (Please Add labels for type of issue)

Description: It's ugly and we are currently using positive/negative buttons, which we might not want. Would prefer to have a cancel button included. 

[//]: # (If an issue is a bug or defect, please list steps to be able to reproduce bug and a screenshot of the defect i.e terminal message)

Story Points: 5

Related Issues: N/A

[//]: # (Notes:)
[//]: # (- Testing and Documentation tasks should always be assumed, but only need to be explicitly stated if there is something that ~~does not~~ need modification. For example,)
[//]: # (some Issues/Stories may not require one of Unit or Intent tests, and this should be indicated in the task list.)
[//]: # (- Code Reviews are also always assumed to be required, but an issue should only indicate in the description if a review is not required, but someone still shouldn't)
[//]: # (merge their own pull request)

### PR DESCRIPTION
**Pull request by gatilogo**
[//]: # (These are comments and are used for reference and will not show up in the PR)

Description: Removed alertdialog and added swipe functionality for edit and delete vibe event. To make this work, the listview was also converted to a recyclerview, which involves having to rewrite the MyVibesAdapter class

[//]: # (Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change)

[//]: # (automatically closes given issue number)
Closes #133 

Proof of Concept

[//]: # (Attach Image of feature added or fix applied and working)
[//]: # (Please provide images of tests passing)
edit
delete

Checklist

[//]: # (Add an x between square brackets to mark item as done)

- [x] Follows style guidelines and is **properly commented**
- [x] Code changes generate no new warnings

### CODE REVIEW

File: app/src/main/java/com/example/vybe/SwipeItemTouchHelper.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import androidx.recyclerview.widget.ItemTouchHelper;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.example.vybe.AddEdit.AddEditVibeEventActivity;
+import com.example.vybe.Models.VibeEvent;
+
+/**
+ * SwipeItemTouchHelper includes the Edit and Delete swipe functionality for a MyVibesAdapter item.
+ * This is attached to the recycler view defined in the MyVibesActivity class.
+ */
```
Comment (raysarinas) [inline app/src/main/java/com/example/vybe/SwipeItemTouchHelper.java]: very good encapsulation, madam

File: app/src/main/java/com/example/vybe/MyVibesActivity.java [hunk 2]
```diff
@@ -361,4 +326,22 @@ public void onRequestPermissionsResult(int requestCode,
             }
         }
     }
+
```
Comment (jacobwong98) [inline app/src/main/java/com/example/vybe/MyVibesActivity.java]: Where da Javadocs at

Comment (raysarinas) [reply app/src/main/java/com/example/vybe/MyVibesActivity.java]: correct!

File: app/src/main/java/com/example/vybe/MyVibesAdapter.java [hunk 3]
```diff
@@ -1,80 +1,88 @@
...
     private static final String TAG = "MyVibesAdapter";
+    private FirebaseFirestore db = FirebaseFirestore.getInstance();
     private Context context;
     private ArrayList<VibeEvent> vibeEventList;
     private int resource;
-    private TextView dateField;
-    private TextView vibeNameField;
-    private ImageView vibeImage;
+
+    public class VibeEventHolder extends RecyclerView.ViewHolder {
```
Comment (jacobwong98) [inline app/src/main/java/com/example/vybe/MyVibesAdapter.java]: Where they at?

File: app/src/main/java/com/example/vybe/MyVibesAdapter.java [hunk 4]
```diff
@@ -1,80 +1,88 @@
...
 
-        return view;
+        holder.itemView.setOnClickListener((View v) -> {
+            Intent viewVibe = new Intent(context, ViewVibeActivity.class);
+            viewVibe.putExtra("vibeEvent", vibeEvent);
+            context.startActivity(viewVibe);
+        });
     }
 
-    /**
```
Comment (jacobwong98) [inline app/src/main/java/com/example/vybe/MyVibesAdapter.java]: Oh no :(

**Pull request merged**